### PR TITLE
Detect capturing in ios

### PR DESCRIPTION
**Pull request by qliqdev**
Added some functions to detect user Capturing Video or Screenshot

### CODE REVIEW

**Comment by inexuscore** [comment]
For now, I'm using the Storage plugin to detect media captures. Whenever the user opens the camera, or switches to the Gallery, etc. I store a flag in the storage, when the app is resumed, I check for that, basic stuff. Would it be possible to support this on both Android and iOS? I mean detecting media captures, any idea when this could be available? Thanks a lot.

**Comment by oneclickwebstudio** [comment]
> For now, I'm using the Storage plugin to detect media captures. Whenever the user opens the camera, or switches to the Gallery, etc. I store a flag in the storage, when the app is resumed, I check for that, basic stuff. Would it be possible to support this on both Android and iOS? I mean detecting media captures, any idea when this could be available? Thanks a lot.

I've researched for Android the same function but could't find how to blur the screen while system triggers screen video capturing

**Comment by robingenz** [comment]
Please merge the `main` branch and resolve the merge conflicts so I can see the changes.

**Comment by oneclickwebstudio** [comment]
> Please merge the `main` branch and resolve the merge conflicts so I can see the changes.
ok

**Comment by oneclickwebstudio** [comment]
> Please merge the `main` branch and resolve the merge conflicts so I can see the changes.

done

**Comment by robingenz** [comment]
Did you forget to push your changes?
There are still merge conflicts.

**Comment by oneclickwebstudio** [comment]
> Did you forget to push your changes? There are still merge conflicts.

Done!

File: ios/Plugin/PrivacyScreenPlugin.swift [hunk 1]
```diff
@@ -36,9 +41,20 @@ public class PrivacyScreenPlugin: CAPPlugin {
...
         self.privacyViewController!.modalPresentationStyle = UIModalPresentationStyle.overFullScreen
+        if !UIAccessibility.isReduceTransparencyEnabled {
+            let blurEffect = UIBlurEffect(style: .dark)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            blurEffectView.frame = self.privacyViewController!.view.bounds
+            blurEffectView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            self.privacyViewController!.view.addSubview(blurEffectView)
+        } else {
+            self.privacyViewController!.view.backgroundColor = .gray
+        }
```
Comment (robingenz) [inline ios/Plugin/PrivacyScreenPlugin.swift:56]: Please make sure that you do not submit multiple features in one PR. This PR should only contain changes that are actually needed for the feature.
Are these changes necessary for detection? If yes, why?

File: ios/Plugin/PrivacyScreenPlugin.swift [hunk 2]
```diff
@@ -50,6 +66,37 @@ public class PrivacyScreenPlugin: CAPPlugin {
...
+            return
+        }
+
+        self.alert = UIAlertController(title: "Attention",
+                                              message: "Screenshots may contain sensitive data. Do not keep them on your phone!",
+                                              preferredStyle: .alert)
+        self.alert?.addAction(UIAlertAction(title: "OK", style: .default))
+        DispatchQueue.main.async {
+            self.bridge?.viewController?.present(self.alert!, animated: false, completion: nil)
+        }
```
Comment (robingenz) [inline ios/Plugin/PrivacyScreenPlugin.swift:97]: The plugin should not interact with the user. 
I propose an event listener instead.
Example: https://github.com/ionic-team/capacitor-plugins/tree/main/app#addlistenerappstatechange-

Everyone should decide on their own how to deal with this event.
For example, an app wants to display the message in a different language.

**Review by robingenz — CHANGES_REQUESTED** [review]

**Comment by robingenz** [comment]
Thank you @oneclickwebstudio. I have reviewed the PR. Please add the `package-lock.json` file again.

File: ios/Plugin/PrivacyScreenPlugin.swift (hunk 1 above)
Comment (oneclickwebstudio) [reply ios/Plugin/PrivacyScreenPlugin.swift:56]: Nope. It is another feature to make blur effect instead of grey background. Last time we decided to make config so user could pass the color of background or blured.

Can i make one method for ex. ```setConfigs``` in frontend, where user could pass options?

File: ios/Plugin/PrivacyScreenPlugin.swift (hunk 2 above)
Comment (oneclickwebstudio) [reply ios/Plugin/PrivacyScreenPlugin.swift:97]: Sure. Also i wanted to pass these strings in configs. where user could define if this alert should appear.

**Comment by oneclickwebstudio** [comment]
> Thank you @oneclickwebstudio. I have reviewed the PR. Please add the `package-lock.json` file again.

Done!

File: ios/Plugin/PrivacyScreenPlugin.swift (hunk 1 above)
Comment (oneclickwebstudio) [reply ios/Plugin/PrivacyScreenPlugin.swift]: If ok can you define method as it should be and i will continue

Comment (robingenz) [reply ios/Plugin/PrivacyScreenPlugin.swift]: Let's discuss this in the other PR (#34), please. I will be happy to make a suggestion for the interface. In this PR the code should not be included.

File: ios/Plugin/PrivacyScreenPlugin.swift [hunk 3]
```diff
@@ -33,23 +35,40 @@ public class PrivacyScreenPlugin: CAPPlugin {
...
 
+    @objc private func didDetectRecording() {
+        guard self.isEnabled else {
+            return
+        }
+
+        if #available(iOS 11.0, *) {
+            if UIScreen.main.isCaptured {
+                DispatchQueue.main.async {
+                    self.bridge?.viewController?.present(self.privacyViewController!, animated: false, completion: nil)
```
Comment (robingenz) [inline ios/Plugin/PrivacyScreenPlugin.swift:63]: The following code is missing at this point or not?
```swift
self.privacyViewController = UIViewController()
self.privacyViewController!.view.backgroundColor = UIColor.gray
self.privacyViewController!.modalPresentationStyle = UIModalPresentationStyle.overFullScreen
```
(copied from line 41)

Here's what I suggest:
1. New method `presentPrivacyViewController`:
```swift
private func presentPrivacyViewController() {
    self.privacyViewController = UIViewController()
    self.privacyViewController!.view.backgroundColor = UIColor.gray
    self.privacyViewController!.modalPresentationStyle = UIModalPresentationStyle.overFullScreen
    DispatchQueue.main.async {
        self.bridge?.viewController?.present(self.privacyViewController!, animated: false, completion: nil)
    }
}
```
2. New method `dismissPrivacyViewController`:
```swift
private func dismissPrivacyViewController() {
       DispatchQueue.main.async {
            self.privacyViewController?.dismiss(animated: false, completion: nil)
        }
}
```

Use these two methods in `onAppWillResignActive`, `onAppDidBecomeActive` and `didDetectRecording`.

**Comment by qliqdev** [comment]
closing this, opened new branch with only capturing feature  in [MR 51](https://github.com/capacitor-community/privacy-screen/pull/51)
